plugins/destination/postgresql/client/deletestale.go

```diff
@@ -9,15 +9,23 @@ import (
 	"github.com/jackc/pgx/v4"
 )
 
+// DeleteStale deletes records where '_cq_source_name' matches the current 'sourceName', and the '_cq_sync_time'
+// is older than the current 'syncTime'.
+// It also deletes records where these fields are null, to avoid old stale records (from previous versions)
+// staying in the database indefinitely.
 func (p *Client) DeleteStale(ctx context.Context, table string, source string, syncTime time.Time) error {
 	var sb strings.Builder
 	sb.WriteString("delete from ")
 	sb.WriteString(pgx.Identifier{table}.Sanitize())
-	sb.WriteString(" where ")
-	sb.WriteString(schema.CqSourceName.Name)
+	sb.WriteString(" where (")
+	sb.WriteString(schema.CqSourceNameColumn.Name)
 	sb.WriteString(" = $1 and ")
-	sb.WriteString(schema.CqSyncTime.Name)
-	sb.WriteString(" < $2")
+	sb.WriteString(schema.CqSyncTimeColumn.Name)
+	sb.WriteString(" < $2) or ")
+	sb.WriteString(schema.CqSourceNameColumn.Name)
+	sb.WriteString(" is null or ")
```

Comment:
why is that ? I wouldn't delete anything that is not 100% ours. if users upgrade to a new version it is up to him to drop the database and re-run in `overwrite-delete-stale` mode. we shouldn't go into edgecase otherwise we can delete user's data.

Reply:
1) This section only runs when the users runs in `overwrite-delete-stale` mode.
2) This code removes the need to drop the database (and, if the user doesn't drop the database, removes resources that can hang there forever)
3) The 'dropping' of the data will only happen **once**. After the next run the behaviour will always be as-expected.